python/lsst/source/injection/inject_engine.py

```diff
@@ -547,6 +589,52 @@ def inject_galsim_objects_into_exposure(
                         err,
                     )
                 continue
+
+            var_template = image_template.copy()
+            var_template *= variance_scale[object_common_bounds]
+
+            if add_noise:
```

Comment:
As this is a fairly substantial piece of code, I'd suggest splitting this off into a new function (`add_noise_to_galsim_exposure`?) and move everything there. This `if` block can then just call that.

Reply:
Agreed.